tutorial/nixify/1_nix_flake.md

```diff
@@ -0,0 +1,223 @@
+
+#### DevShells
+
+- This puts you in a development shell with only the packages you need.
+- You can define how your shell will look using `pkgs.mkShell`.
+- `pkgs.mkShell` creates a derivation that is evaluated by the command `nix develop`
+- Derivation given by `devShells.${system}.default` is evaluated by default, you can also define custom `devShell` like `devShells.${system}.mydevShell` and run it using `nix develop .#mydevShell`
+
+It's time to now run the following commands and see the `flake.nix` in action:
+- `nix build`
+- `nix run`
+- `nix develop`
+
+## Nixify Haskell package
+
+We saw how to use `pkgs.hello` from `nixpkgs` in the basic example above, now it is time to define build instructions to build a package for the haskell project `todo-app`.\
+Tl;dr Here's the `flake.nix` for this section:
+```nix
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  };
+  outputs = { self, nixpkgs }:
+    let
+      system = "aarch64-darwin";
+      pkgs = nixpkgs.legacyPackages.${system};
+      overlay = final: prev: {
+        todo-app = final.callCabal2nix "todo-app" ./. { };
+      };
+      myHaskellPackages = pkgs.haskellPackages.extend overlay;
+    in
+    {
+      packages.${system}.default = myHaskellPackages.todo-app;
+      apps.${system}.default = {
+        type = "app";
+        program = "${self.packages.${system}.default}/bin/todo-app";
+      };
+    };
+}
+```
+Let's break it down!
+### haskellPackages
+
+This is an attribute set that consists of all the haskell projects maintained in Nixpkgs. Since our local project will not be part of this, we have to add it. Technically you can just use  `packages.${system}.default = pkgs.${system}.haskellPackages.callCabal2nix "todo-app" ./. { };` but adding it to `haskellPackages` keeps it all in one place and consuming flakes will not have to think about where to find this package. 
+
+### Overlay
+
+This is a function that will output an attribute consisting of the packages that you want to either add into `haskellPackages` or replace if they already exist. It takes two parameters, first parameter being the final state of `haskellPackages` after applying all the overrides and the second being the state of `haskellPackages` before applying the overrides.
+
+### callCabal2nix
+
+A function that takes the `name` of package (`todo-app`), `src` of cabal file (`./.`, i.e current directory), and overrides (`{ }`, i.e none) as parameters and returns a derivation that will build your package. This function internally uses ["cabal2nix"](https://github.com/NixOS/cabal2nix), a Haskell tool that generates Nix build instructions from a cabal file.
+
+### Time to run!
+After replacing the previous `flake.nix` with the current one, run the following:
+- `nix build` builds the haskell project and symlinks your executable in `./result/bin`
+- `nix run` builds (if not already) and runs the `todo-app` executable
```

Comment:
@shivaraj-bh 

Perhaps we should embed [asciinema](https://asciinema.org/) screencasts here for these commands? Also for the `hello` commands above?

Reply:
We will have a video tutorial accompanying this blog, that will resolve this!